test/rekt/features/apiserversource/data_plane.go

```diff
@@ -351,6 +352,151 @@ func SendsEventsForAllResources() *feature.Feature {
 	return f
 }
 
+func SendsEventsForAllResourcesWithNamespaceSelector() *feature.Feature {
+	source := feature.MakeRandomK8sName("apiserversource")
+	sink := feature.MakeRandomK8sName("sink")
+	f := feature.NewFeatureNamed("Send events for all resources within select namespaces")
+
+	f.Setup("install sink", eventshub.Install(sink, eventshub.StartReceiver))
+
+	sacmName := feature.MakeRandomK8sName("apiserversource")
+	f.Setup("Create Service Account for ApiServerSource with RBAC for v1.Pod resources",
+		setupAccountAndRoleForPods(sacmName))
+
+	testNS1 := feature.MakeRandomK8sName("source-namespace")
+	testNS2 := feature.MakeRandomK8sName("source-namespace")
+	testNS3 := feature.MakeRandomK8sName("source-namespace")
+
+	// create two new namespaces with matching selector
+	f.Setup("create test-a namespace with matching selector", namespace.Install(testNS1, namespace.WithLabels(map[string]string{"env": "development"})))
+	f.Setup("create test-a namespace with matching selector", namespace.Install(testNS2, namespace.WithLabels(map[string]string{"env": "development"})))
+
+	// create one new namespace that doesn't match selector
+	f.Setup("create test-a namespace with matching selector", namespace.Install(testNS3, namespace.WithLabels(map[string]string{"env": "production"})))
+
+	cfg := []manifest.CfgFn{
+		apiserversource.WithServiceAccountName(sacmName),
+		apiserversource.WithEventMode("Reference"),
+		apiserversource.WithSink(svc.AsKReference(sink), ""),
+		apiserversource.WithResources(v1.APIVersionKindSelector{
+			APIVersion: "v1",
+			Kind:       "Pod",
+		}),
+		apiserversource.WithNamespaceSelector(&metav1.LabelSelector{
+			MatchLabels: map[string]string{"env": "development"},
+		}),
+	}
+
+	f.Setup("install ApiServerSource", apiserversource.Install(source, cfg...))
+	f.Setup("ApiServerSource goes ready", apiserversource.IsReady(source))
+
+	pod1 := feature.MakeRandomK8sName("example")
+	pod2 := feature.MakeRandomK8sName("example")
+	pod3 := feature.MakeRandomK8sName("example")
```

Comment:
I'd append a fixed suffix to the prefix based on the variable name to make it more predictable and clear

Reply:
Fixed